core/src/utils/overlays.ts

```diff
@@ -327,31 +329,8 @@ export const getOverlays = (doc: Document, selector?: string): HTMLIonOverlayEle
     .filter(c => c.overlayIndex > 0);
 };
 
-/**
- * Gets the top-most/last opened
- * overlay that is currently presented.
- */
-const getTopOpenOverlay = (doc: Document): HTMLIonOverlayElement | undefined => {
-  const overlays = getOverlays(doc);
-  for (let i = overlays.length - 1; i >= 0; i--) {
-    const overlay = overlays[i];
-
-    /**
-     * Only consider overlays that
-     * are presented. Presented overlays
-     * will not have the .overlay-hidden
-     * class on the host.
-     */
-    if (!overlay.classList.contains('overlay-hidden')) {
-      return overlay;
-    }
-  }
-
-  return;
-}
-
 export const getOverlay = (doc: Document, overlayTag?: string, id?: string): HTMLIonOverlayElement | undefined => {
-  const overlays = getOverlays(doc, overlayTag);
+  const overlays = getOverlays(doc, overlayTag).filter(o => !isOverlayHidden(o));
   return (id === undefined)
     ? overlays[overlays.length - 1]
     : overlays.find(o => o.id === id);
```

Comment:
If you try to find an overlay by ID and it's closed, you'll get `undefined`, which feels weird 🤔 Maybe only the `overlays[overlays.length - 1]` should be filtered to non-hidden overlays?

Reply:
That's really good feedback. Implementers may have a use case of getting an overlay by id before it's presented. I made that change and also added an e2e test.